src/python/pants/engine/console.py

```diff
@@ -2,14 +2,52 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 import sys
+from typing import Any, Optional
 
 from colors import blue, green, red
 
+from pants.engine.native import Native
+
+
+#TODO this needs to be a file-like object/stream
+class NativeWriter:
+  def __init__(self, session: Any):
+    self._native = Native()
+    self._session = session._session
+
+  def write(self, payload: str):
+    raise NotImplementedError
+
+  def flush(self):
+    #I dunno what if anything needs to happen here
```

Comment:
It depends on how we hold the output in EngineDisplay. If we buffer stdout/err, we may want to force-flush it whenever this function gets called, via an FFI call. For now, adding a TODO with that reasoning would be fine :)